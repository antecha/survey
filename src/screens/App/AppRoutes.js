import React from 'react';
import { Route } from 'react-router-dom';
import { Layout } from 'antd';
import CreateSurvey from '../CreateSurvey';

const { Content } = Layout;

export const AppContent = ({ children }) => (
  <Content
    style={{
      margin: '24px 16px',
      padding: 24,
      background: '#fff',
      minHeight: 280,
    }}
  >
    {children}
  </Content>
);

const AppRoutes = () => {
  return (
    <React.Fragment>
      <Route exact path="/">
        <AppContent>
          <CreateSurvey />
        </AppContent>
      </Route>

      <Route path="/create-survey">
        <AppContent children="create survey" />
      </Route>
    </React.Fragment>
  );
};

export default AppRoutes;
