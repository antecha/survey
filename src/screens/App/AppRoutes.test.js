import React from 'react';
import { create } from 'react-test-renderer';
import AppRoutes, { AppContent } from './AppRoutes';
import { MemoryRouter } from 'react-router';

jest.mock('antd', _ => ({
  ...require.requireActual('antd'),
  Tabs: () => <div />,
}));

describe('AppContent', () => {
  const testRenderer = create(
    <AppContent>
      <div />
    </AppContent>,
  );

  it('renders', () => {
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
});

describe('AppRoutes', () => {
  const testRenderer = create(
    <MemoryRouter>
      <AppRoutes />,
    </MemoryRouter>,
  );

  it('renders', () => {
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
});
