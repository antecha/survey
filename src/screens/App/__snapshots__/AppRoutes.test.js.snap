// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppContent renders 1`] = `
<main
  className="ant-layout-content"
  style={
    Object {
      "background": "#fff",
      "margin": "24px 16px",
      "minHeight": 280,
      "padding": 24,
    }
  }
>
  <div />
</main>
`;

exports[`AppRoutes renders 1`] = `
<main
  className="ant-layout-content"
  style={
    Object {
      "background": "#fff",
      "margin": "24px 16px",
      "minHeight": 280,
      "padding": 24,
    }
  }
>
  <div>
    <div />
  </div>
</main>
`;
